pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,100 @@
+package bmc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// contains tells whether the array of redfish.VirtualMediaType
+// has the element redfish.VirtualMediaType.
+func contains(array []redfish.VirtualMediaType, element redfish.VirtualMediaType) bool {
+	for _, n := range array {
+		if element == n {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool, httpClient *http.Client) error {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint:   address,
+		Username:   user,
+		Password:   password,
+		Insecure:   insecure,
+		HTTPClient: httpClient,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		if strings.Contains(err.Error(), "i/o timeout") {
+			return &AccessVirtualMediaValidationError{message: "timeout reached when trying to access redfish BMC"}
+		}
+		return &AccessVirtualMediaValidationError{message: "could not connect to redfish BMC"}
+	}
+	defer client.Logout()
+
+	service := client.Service
+
+	availableSystems, err := service.Systems()
+	if err != nil {
+		return err
+	}
+	if len(availableSystems) == 0 {
+		return fmt.Errorf("could not retrieve Systems")
```

Comment:
Let's be specific: "no Systems are found"

Reply:
done